docs/rfcs/IntelGpuDialect.md

```diff
@@ -0,0 +1,122 @@
+# RFC: Intel GPU Dialect (I-GPU)
+
+Nishant Patel
+
+## Summary
+
+We propose the Intel Gpu Dialect as a custom Intel dialect to unlock the underlying runtimes (Level zero & Sycl) best suited for Intel Gpus.
+This Dialect will be at a lower level than the upstream GPU Dialect.
```

Comment:
I think it should be pointed out that 
1) Intel GPU dialect is a conversion target for upstream GPU dialect as an implementation dialect.
2) Goal is not to provide a lower level alternative to all GPU dialect ops and provide just things lacking in upstream GPU dialect.
But on the other hand may be too much information for a summary.

Reply:
From the description, it is very clear what proposed is not a full set of dialect, just an extension to upstream GPU dialect.    If possible, we would like to upstream it.